distributed/tests/test_worker.py

```diff
@@ -895,7 +942,7 @@ def __sizeof__(self):
 
     assert future.status == "error"
 
-    with pytest.raises(TypeError):
+    with pytest.raises(Exception):
```

Comment:
please make it narrower and add a match

Reply:
In the test name and/or docstring, please clarify that this is a key that is immediately stored to disk

Please add a test for a key fails to pickle and that is NOT immediately stored to disk.